### PR TITLE
Add forest plot for panel 4G

### PR DESCRIPTION
**Pull request by jharenza**
#### What scientific question is your analysis addressing?
Add HGG subtypes forest plot, staggered on #1332 

#### What was your approach?
Add forest plot, including reference subtype

#### What GitHub issue does your pull request address?
Ahh, I guess I never made an issue for this..

Upate, found it #1323

#### Which areas should receive a particularly close look?
NA

#### Is there anything that you want to discuss further?
From #1032 
>Pairwise comparisons show a signficant difference between “DMG, H3 K28, TP53 loss” and “DMG, H3 K28”, as well as between “HGG, H3 wildtype” and “DMG, H3 K28”

Since DMG H3 K28 is the reference, why isn't the p value for HGG, H3 wildtype significant in this additive model? It would actually be nice if we can show this significance value on the FP - maybe we can recode that one grey and add the pairwise pvalue? 😬 

#### Reproducibility Checklist

- [ ] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [ ] This analysis has been added to continuous integration.

#### Documentation Checklist

- [ ] This analysis module has a `README` and it is up to date.
- [ ] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [ ] The analytical code is documented and contains comments.

### CODE REVIEW

File: figures/scripts/fig4-hgg-subtype-forest-plot.R [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+
+# Get n and event info from glance output
+survival_n <- broom::glance(survival_result) %>%
+  select(n, nevent)
+
+# Convert survival model result to data frame, and exponentiate estimates/CIs to get HRs
+survival_df <- broom::tidy(survival_result) %>%
+  # Add DMG, H3 K28 as reference
+  add_row(term = "molecular_subtypeDMG, H3 K28",
```
Comment (sjspielman) [inline figures/scripts/fig4-hgg-subtype-forest-plot.R:57]: 👍

**Review by sjspielman — COMMENTED** [review]
Looks good! One other quick thought - I think it's great that you added the reference level here into the plot. I wonder if we should change it's placement and/or color?
Specifically, I think it makes sense for the reference to be the bottom row of the plot (though, still next to the other HGG levels). Also, the shape is white but it's not actually "not significant" - it's just not anything. What do you think about either making it gray, or maybe changing the shape to not be a diamond? Basically, some kind of visual cue that it's a different kind of point.

**Comment by jharenza** [comment]
>Looks good! One other quick thought - I think it's great that you added the reference level here into the plot. I wonder if we should change it's placement and/or color?
Specifically, I think it makes sense for the reference to be the bottom row of the plot (though, still next to the other HGG levels). Also, the shape is white but it's not actually "not significant" - it's just not anything. What do you think about either making it gray, or maybe changing the shape to not be a diamond? Basically, some kind of visual cue that it's a different kind of point.

YES! I also had the thought about grey or another shape, and bottom is fine with me. I have multiple refs in other forest plots because of the groupings, but we can add those to the bottom or bottom of the group - what do you think?

**Comment by sjspielman** [comment]
Ha, I just edited that comment because I realize I missed you had already suggested the gray color!

I think at the bottom of each _grouping_ makes the most sense.

**Comment by jharenza** [comment]
> Ha, I just edited that comment because I realize I missed you had already suggested the gray color!
> 
> I think at the bottom of each _grouping_ makes the most sense.

updated

**Review by sjspielman — APPROVED** [review]
I made two quick changes here: 
1) plot export size, and 
2) added a variable `ref_term` since that string is reused.

This does not require CI, so I am going to go ahead and approve and merge this into `surv-hgg-subtype` now in the interest of time.

**Pull request merged**